Support/lib/ruby_utils.rb

```diff
@@ -0,0 +1,55 @@
+module RubyUtils
+  module_function
+
+  # Try to find the Ruby executable `name`. Return an array containing the
+  # command(s) needed to run the executable, or `nil` if the executable could
+  # not be found.
+  #
+  # Typical usage looks like this:
+  #
+  #     Dir.chdir(ENV['TM_PROJECT_DIRECTORY']) if ENV['TM_PROJECT_DIRECTORY']
+  #     executable = RubyUtils.find_executable('rspec')
+  #     if executable
+  #       system(*executable, '--some', '--additional', 'args')
+  #     else
+  #       # Executable not found, so use fallback / display alert / …
+  #     end
+  #
+  # Supports the following cases:
+  #
+  # 1. If an appropriate `TM_*` environment variable is present and points to an
```

Comment:
> We can add a “project scope” for projects with a Gemfile so that we can target these (or target projects without), here’s the list of current project attributes: https://github.com/textmate/textmate/blob/master/Frameworks/DocumentWindow/src/DocumentController.mm#L1429

How would that work?

> Not having it trump everything means that if a user reports that for some project, our logic is wrong, we cannot provide them with any workaround, they’ll have to wait for someone to fix the code and push a new release, that’s IMO where the environment variable comes in: to allow users to fix things locally.

Not sure I understand. By default it would use what's in the Gemfile, the user has already control of the Gemfile.